### PR TITLE
Changed assertion in library_async so handleSleepReturnValue can be a bool (fix #13359)

### LINKED ISSUE #13359: EM_JS can't return a bool because of bogus assertion check
```
handleSleepReturnValue  = 1;

assert(!handleSleepReturnValue || typeof handleSleepReturnValue === 'number');

// !handleSleepReturnValue || typeof handleSleepReturnValue === 'number'    == false; 

// condition == false;
function assert(condition, text) {
  // Now this cause it to become true
  if (!condition) {
    // which cause to throw an error
    abort('Assertion failed: ' + text);
  }
}




```

### PR DESCRIPTION
**Pull request by bpuzycki**

### CODE REVIEW

**Comment by bpuzycki** [comment]
fixed #13359

**Comment by bpuzycki** [comment]
Getting a many errors related to "unexpected metadata" on the circleCI tests. Anybody know what this could be about given the small change I made?

**Comment by kripken** [comment]
Thanks! Code looks good.

For a test, I think you can extend `tests/browser/async_returnvalue.cpp, tests/browser/async_returnvalue.js` to add another function that returns a boolean. To test it locally, run `./tests/runner.py browser.test_async_returnvalue`.

**Comment by kripken** [comment]
Test looks good @bpuzycki but it fails on
```
Uncaught import sync_tunnel_bool was not in ASYNCIFY_IMPORTS, but changed the state
```
You need to modify the python file to add that to the flags passed in.

Please also merge in latest `main` to here, as that will fix some unrelated errors from older `main` that are present.

File: tests/test_browser.py [hunk 1]
```diff
@@ -3292,7 +3292,7 @@ def test_async_iostream(self):
   # ASYNCIFY_IMPORTS.
   # To make the test more precise we also use ASYNCIFY_IGNORE_INDIRECT here.
   @parameterized({
-    'normal': (['-s', 'ASYNCIFY_IMPORTS=[sync_tunnel]'],), # noqa
+    'normal': (['-s', 'ASYNCIFY_IMPORTS=[sync_tunnel, sync_tunnel_bool]'],), # noqa
     'response': (['-s', 'ASYNCIFY_IMPORTS=@filey.txt'],), # noqa
     'nothing': (['-DBAD'],), # noqa
     'empty_list': (['-DBAD', '-s', 'ASYNCIFY_IMPORTS=[]'],), # noqa
```
Comment (kripken) [inline tests/test_browser.py:3298]: I think it also needs to be added on line 3303 (there is a little duplication in this test, which is not great I admit...)

Comment (bpuzycki) [reply tests/test_browser.py:3298]: Yup, am adding that now. Thanks for the quick response.

**Review by kripken — APPROVED** [review]
Perfect, thanks!

**Pull request merged**